ark/type/src/parser/tuple.ts

```diff
@@ -234,6 +245,8 @@ export type inferTupleExpression<
 	args
 > = def[1] extends "[]"
 	? inferDefinition<def[0], $, args>[]
+	: def[1] extends "?"
```

Comment:
This should be in `inferTupleLiteral` as a special case, not `inferTupleExpression`